sycl/source/detail/usm/usm_impl.cpp

```diff
@@ -44,12 +32,31 @@ __SYCL_INLINE_VER_NAMESPACE(_V1) {
 using alloc = sycl::usm::alloc;
 
 namespace detail {
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+extern xpti::trace_event_data_t *GSYCLGraphEvent;
+#endif
 namespace usm {
 
 void *alignedAllocHost(size_t Alignment, size_t Size, const context &Ctxt,
                        alloc Kind, const property_list &PropList,
                        const detail::code_location &CodeLoc) {
-  XPTI_CREATE_TRACEPOINT(CodeLoc);
+#ifdef XPTI_ENABLE_INSTRUMENTATION
+  // Stash the code location information and propagate
+  detail::tls_code_loc_t CL(CodeLoc);
+  XPTIScope PrepareNotify((void *)alignedAllocHost,
+                          (uint16_t)xpti::trace_point_type_t::node_create,
```

Comment:
Are those cast absolutely necessary? If so, can we modify the ctor so that caller could avoid them (extra overload maybe?)